src/pages/experiments/[experimentId]/plots-and-tables/embedding-categorical/index.jsx

```diff
@@ -57,26 +57,55 @@ const EmbeddingCategoricalPlot = () => {
 
   const router = useRouter();
   const { experimentId } = router.query;
+  const [plotSpec, setPlotSpec] = useState({});
 
+  // First, load experiment ID data from the router.
   useEffect(() => {
-    if (isBrowser) {
-      if (!data) {
-        dispatch(loadEmbedding(experimentId, embeddingType));
-      }
-      dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
-      dispatch(loadCellSets(experimentId));
+    if (!experimentId || !isBrowser) {
+      return;
+    }
```

Comment:
If we don't check for whether experimentId exists, this will start loading embeddings/plot configs/cell sets with experimentId undefined, which is a bug.